### ISSUE TITLE
[bug] Search Engine does not allow to search by email field. If selected 'Begin With'

### ISSUE THREAD

**Issue by z0lo13** [YetiForceCompany/YetiForceCRM#13960]
# 🐞 bug report

### 🔥 How to trigger the error

1. Go to settings ->Search-> Add email field to be used for search in any module, for example Leads.
2. Wait until Search Labels generated.
3. Go to Leads Module select 'Begin With' and try to search by email.
4. See error:

### 👎 Actual Behavior

Search not Working by email field

### 👍 Expected Behavior

Search Working by email field

### 👷 Testing

- [✔️] I confirm that I tested it at https://gitstable.yetiforce.com
- [✔️] I confirm that I tested it at https://gitdeveloper.yetiforce.com

### 📷 Screenshot of configuration

![ConfReport](https://user-images.githubusercontent.com/23143958/102798278-dc8b7780-437e-11eb-81c8-e45a195b3fc8.png)

### 📝 PHP/Apache/Nginx/Browser/CRM Logs

5. ```Next yii\db\Exception: SQLSTATE[42000]: Syntax error or access violation: 1064 syntax error, unexpected '@', expecting $end
The SQL being executed was: SELECT `csl`.`crmid`, `csl`.`setype`, `csl`.`searchlabel`, MATCH(csl.searchlabel) AGAINST('z0lo@me.com*' IN BOOLEAN MODE) AS `matcher` FROM `u_yf_crmentity_search_label` `csl` INNER JOIN `vtiger_tab` ON csl.setype = vtiger_tab.name INNER JOIN `vtiger_leaddetails` ON csl.crmid = vtiger_leaddetails.leadid WHERE (MATCH(csl.searchlabel) AGAINST('z0lo@me.com*' IN BOOLEAN MODE)) AND ((`vtiger_tab`.`presence`=0) AND (`csl`.`setype`='Leads') AND (`vtiger_leaddetails`.`converted`=0) AND (`csl`.`userid` LIKE '%,5,%')) ORDER BY `matcher` DESC LIMIT 100 in vendor/yetiforce/yii2/db/Schema.php:678
Stack trace:
#0 vendor/yetiforce/yii2/db/Command.php(1304): yii\db\Schema->convertException(Object(PDOException), 'SELECT `csl`.`c...')
#1 vendor/yetiforce/yii2/db/Command.php(1165): yii\db\Command->internalExecute('SELECT `csl`.`c...')
#2 vendor/yetiforce/yii2/db/Command.php(407): yii\db\Command->queryInternal('fetchAll', NULL)
#3 vendor/yetiforce/yii2/db/Query.php(248): yii\db\Command->queryAll()
#4 app/RecordSearch.php(68): yii\db\Query->all()
#5 modules/Vtiger/models/Record.php(760): App\RecordSearch->search()
#6 modules/Vtiger/views/BasicAjax.php(121): Vtiger_Record_Model::getSearchResult('z0lo@me.com', 'Leads', 100, 'FulltextBegin')
#7 app/Controller/ExposeMethod.php(65): Vtiger_BasicAjax_View->showSearchResults(Object(App\Request))
#8 app/Controller/ExposeMethod.php(78): Vtiger_BasicAjax_View->invokeExposedMethod('showSearchResul...', Object(App\Request))
#9 include/main/WebUI.php(189): Vtiger_BasicAjax_View->process(Object(App\Request))
#10 index.php(28): Vtiger_WebUI->process(Object(App\Request))
#11 public_html/index.php(11): require('/home/crm/publi...')
#12 {main}
Additional Information:
Array
(
    [0] => 42000
    [1] => 1064
    [2] => syntax error, unexpected '@', expecting $end
)```
